src/comps/rf.cc

```diff
@@ -0,0 +1,176 @@
+#include <lapack.hh>
+#include <RandBLAS.hh>
+#include <RandLAPACK.hh>
+
+#define USE_QR
+#define COND_CHECK
+#define VERBOSE
+
+namespace RandLAPACK::comps::rf {
+
+template <typename T>
+void RF<T>::rf1(
+    int64_t m,
+    int64_t n,
+    const std::vector<T>& A,
+    int64_t k,
+    std::vector<T>& Q, // n by k
+    bool use_qr
+){
+    printf("HI, I AM REGULAR RF. \n");
+
+    using namespace blas;
+    using namespace lapack;
+
+    // Get the sketching operator Omega
+    std::vector<T> Omega(n * k, 0.0);
+    T* Q_dat = Q.data();
+
+    RF::RS_Obj.call(m, n, A, k, Omega);
+
+    // Q = orth(A * Omega)
+    gemm<T>(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, k, n, 1.0, A.data(), m, Omega.data(), n, 0.0, Q_dat, m);
+
+
+    //char name_5[] = "Gemm result";
+    //RandBLAS::util::print_colmaj(m, k, Q_dat, name_5);
+
+    if (RF::cond_check)
+    {
+        // Copy to avoid any changes
+        std::vector<T> Q_cpy (m * k, 0.0);
+        std::vector<T> s(k, 0.0);
+        T* Q_cpy_dat = Q_cpy.data();
+        T* s_dat = s.data();
+
+        lacpy(MatrixType::General, m, k, Q_dat, m, Q_cpy_dat, m);
+        gesdd(Job::NoVec, m, k, Q_cpy_dat, m, s_dat, NULL, m, NULL, k);
+        T cond_num = *s_dat / *(s_dat + k - 1);
```

Comment:
I'm not a fan of this pointer arithmetic. It would be better to just do ``s_dat[0] / s_dat[k - 1]``.